### ISSUE TITLE
Feature request - PIN protection

### ISSUE THREAD

**Issue by lenisko** [raulhaag/MiMangaNu#224]
How about adding simple PIN protection to application

**Comment by Dostoi** [comment]
Why not using an app for that ?
A quick search on f-droid show several results for that.

**Comment by lenisko** [comment]
It's really easy to implement this from dev side.
More apps = more RAM usage, more disk usage, more bloat.

**Comment by Dostoi** [comment]
Yes but asking every dev of every app that you use is kind of a waste of time for them as there's already dev working on app that can do it for all the app you're using. And pin lock isn't a central feature in most app.
But I get your point. In the end it's up to ralhaag anyway.

**Comment by lenisko** [comment]
Not every app needs PIN protection.
Why would you need protection to application which can check weather outside?:)
In MiMangaNu, some people might have stuff which shouldn't be seen by other persons, or kids.

It's just proposal, so yes it's up to @raulhaag :)

**Comment by raulhaag** [comment]
( ͡° ͜ʖ ͡°)

**Comment by raulhaag** [comment]
thinking about it a pair of question come to me: 
- pretect the start (that can be easy done) but if you let the app openned can be accesable 
- or look for a pasword every activity resume
- the images can be seen on the sd 

if any user whit a basic idea of android can see all whitout a problem and i can't encrypt all because the energy consumition go to be too high
not to convinced about it

**Comment by lenisko** [comment]
@raulhaag 
> * pretect the start (that can be easy done) but if you let the app openned can be accesable

Main functionality, it's needed.
> * or look for a pasword every activity resume

Yes, adding option for that in settings would be great. Also catching this event should be very easy for you.
> * the images can be seen on the sd 

You just solved that in this enhancement (thanks) https://github.com/raulhaag/MiMangaNu/issues/220. Images should not be touched, someone might want to download them using your application, and view using other application. With functionality to read & delete we might have clean folder.

> if any user whit a basic idea of android can see all whitout a problem and i can't encrypt all because the energy consumition go to be too high
not to convinced about it

Well, not everyone know how dig into files and like I told, images can be deleted. We want hide covers (cache is without file extension, lets leave it like that) and content. Encrypting everything in this case is not needed (imo)

**Comment by xken2x** [comment]
Some people have h... ( ͡° ͜ʖ ͡°)

Its impractical, only use a external app

**Comment by lenisko** [comment]
@xken2x 
https://github.com/raulhaag/MiMangaNu/issues/221 related a bit :)

**Comment by sm7297** [comment]
Or you can just create separate tab or catagory for those manga you wanted password protect then move those manga into the category or tab if you wish to see tap on the category password will be asked thats it

**Comment by lenisko** [comment]
@sm7297 This ia more complex than enabling PIN in 2 or 3 cases.

**Comment by null-dev** [comment]
_*nudge nudge, hint hint*_
https://github.com/Nightonke/BlurLockView

**Comment by null-dev** [comment]
I can do a PR if raulhaag will approve this...

**Comment by raulhaag** [comment]
@null-dev PR? idon't know what is that, but if not overload the app and i don't have to do nothing is ok

**Comment by lenisko** [comment]
@raulhaag it's pull request, so yeah.. you just have to merge it after job is done, check, and deploy.
